arrow/src/compute/kernels/comparison.rs

```diff
@@ -3832,6 +3835,27 @@ mod tests {
         ilike_utf8,
         vec![true, true, true, false, false, true, false]
     );
+    test_utf8!(
+        test_utf8_array_ilike_tricky_chars,
+        vec!["i", "I", "山"],
+        vec!["I", "i", "山"],
+        ilike_utf8,
+        vec![true, true, true]
+    );
+    test_utf8!(
+        test_utf8_array_ilike_tricky_chars2,
+        vec!["SS", "ß"],
+        vec!["ß", "SS"],
+        ilike_utf8,
+        vec![true, true]
+    );
+    test_utf8!(
+        test_utf8_array_ilike_tricky_chars3,
+        vec!["ı", "İ"],
+        vec!["I", "i"],
+        ilike_utf8,
+        vec![true, true]
+    );
```

Comment:
These two tests fail. same as with the non-builder variant in master. so this doesn't seem to get us around the edge cases in https://www.unicode.org/Public/UCD/latest/ucd/SpecialCasing.txt, sadly